module_utils/network_lsr/argument_validator.py

```diff
@@ -1611,7 +1637,7 @@ def _validate_post(self, value, name, result):
                         "invalid 'interface_name' '%s'" % (result["interface_name"]),
                     )
             else:
-                if not result.get("mac"):
+                if not result.get("mac") and not result.get("match"):
```

Comment:
Please also add a unit tests that asserts the correct behavior for `interface_name` based on whether match.path is set/partially set etc:


Considering that `mac` is not specified:
```
match not specified -> interface_name should be profile name

match: # interface_name should be profile name

match: # interface_name should be profile name
  path: []

match: # interface_name should be unset
  path:
    - example_path
```

Please also consider other scenarios that I might have missed. Thanks.

Reply:
Unit tests were added. Thanks